subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/RepositoryChainModuleSource.java

```diff
@@ -17,7 +17,7 @@
 
 import org.gradle.internal.component.model.ModuleSource;
 
-class RepositoryChainModuleSource implements ModuleSource {
+public class RepositoryChainModuleSource implements ModuleSource {
```

Comment:
:x: Looks like a code smell. You shouldn't have to know what kind of source you use. Or, at least, not an explicit type. Either add an interface `RepositoryAwareModuleSource`, or, make it part of `ModuleSource`.

Reply:
fixed